app/scripts/tmx_products.py

```diff
@@ -0,0 +1,192 @@
+            'extensions', 'mail', 'mobile', 'netwerk', 'other-licenses',
+            'security', 'services', 'suite', 'toolkit', 'webapprt'
+        ]
+        dirs = filter(lambda x: x in dirs_locale, dirs_reference)
+
+    return dirs
+
+
+def create_tmx_content(reference_repo, locale_repo, dirs):
+    ''' Extract strings from repository, return them as a list of PHP array
+        elements. '''
+    tmx_content = []
+    for directory in dirs:
+        path_reference = os.path.join(reference_repo, directory)
+        path_locale = os.path.join(locale_repo, directory)
+
+        rcsClient = silme.io.Manager.get('file')
+        try:
+            l10nPackage_reference = rcsClient.get_package(
+                path_reference, object_type='entitylist')
+        except Exception as e:
+            print 'Silme couldn\'t extract data for', path_reference
+            print e
+            continue
+
+        if not os.path.isdir(path_locale):
+            # Folder doesn't exist for this locale, don't log a warning,
+            # just continue to the next folder.
+            continue
+
+        try:
+            l10nPackage_locale = rcsClient.get_package(
+                path_locale, object_type='entitylist')
+        except Exception as e:
+            print 'Silme couldn\'t extract data for', path_locale
+            print e
+            continue
+
+        strings_reference = {}
+        strings_locale = {}
+        get_strings(l10nPackage_reference, directory, strings_reference)
+        get_strings(l10nPackage_locale, directory, strings_locale)
+        for entity in strings_reference:
+            # Append string to tmx_content, using the format of a PHP array
+            # element
+            translation = escape(
+                strings_locale.get(entity, '')).encode('utf-8')
+            tmx_content.append("'{0}' => '{1}'".format(
+                entity.encode('utf-8'), translation))
+
+    return tmx_content
+
+
+def write_php_file(filename, tmx_content):
+    ''' Write TMX content as a PHP array on file '''
+    target_locale_file = open(filename, 'w')
+    target_locale_file.write('<?php\n$tmx = [\n')
+    for line in tmx_content:
+        target_locale_file.write(line + ',\n')
+    target_locale_file.write('];\n')
+    target_locale_file.close()
+
+
+def main():
+    # Read command line input parameters
+    parser = argparse.ArgumentParser()
+    parser.add_argument('locale_repo', help='Path to locale files')
+    parser.add_argument('reference_repo', help='Path to reference files')
+    parser.add_argument('locale_code', help='Locale language code')
+    parser.add_argument('reference_code', help='Reference language code')
+    parser.add_argument('repository', help='Repository name')
+    args = parser.parse_args()
+
+    dirs = create_directories_list(
+        args.reference_repo, args.locale_repo, args.repository
+    )
+    tmx_content = create_tmx_content(
+        args.reference_repo, args.locale_repo, dirs)
```

Comment:
It doesn't work in Python, you need a parenthesis to go on a new line (this is also the result of formatting for PEP8) :-\